app-frontend/src/app/components/pages/project/analyses/analyses/index.js

```diff
@@ -1,11 +1,208 @@
 import tpl from './index.html';
+import _ from 'lodash';
 
 class AnalysesListController {
+    constructor(
+        $rootScope, $scope, $state, $log, APP_CONFIG,
+        mapService, projectService, analysisService, paginationService,
+        authService, modalService
+    ) {
+        'ngInject';
+        $rootScope.autoInject(this, arguments);
+    }
 
+    $onInit() {
+        this.visible = new Set([]);
+        this.tileServer = `${this.APP_CONFIG.tileServerLocation}`;
+        this.syncMapLayersToVisible();
+        this.fetchPage();
+    }
+
+    $onDestroy() {
+        this.getMap().then((map) => {
+            map.deleteLayers('Project Analyses');
+        });
+    }
+
+    getMap() {
+        return this.mapService.getMap('project');
+    }
+
+    showFirstAnalysis() {
+        this.visible.clear();
+        this.visible.add(this.itemList[0].id);
+        this.syncMapLayersToVisible();
+    }
+
+    showAllAnalyses() {
+        this.itemList.forEach(item => {
```

Comment:
Could we make this a bit simpler?
```suggestion
        this.visible = new Set(this.itemList);
```

Reply:
This won't work since we'd need to pick the `id` property. Disregard!